### PR TITLE
td9 - functionality to submit new comments

### PR DESCRIPTION
**Pull request by SFagelman**

### CODE REVIEW

File: src/components/AddComment.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      <textarea
+        name="textbox"
+        id="new-comment-textbox"
+        cols="30"
+        rows="3"
+        placeholder="Add comment..."
+        onChange={(event) => {
+          setNewComment(event.target.value);
+        }}
+      ></textarea>
```
Comment (douglashellowell) [inline src/components/AddComment.jsx]: if you want to create a controlled component don't forget to reflect the value that's held in state back into the input!

`value={newComment}`

File: src/components/AddComment.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  const { loggedInUser } = useContext(UserContext);
+
+  const [newComment, setNewComment] = useState("");
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    addCommentByArticle(article_id, loggedInUser.username, newComment);
+    setNewComment("");
+    let commentInput = document.getElementById("new-comment-textbox");
+    commentInput.value = "";
```
Comment (douglashellowell) [inline src/components/AddComment.jsx]: you shouldn't be doing any manual DOM manipulation in React!

**Comment by douglashellowell** [comment]
🚨  you have an infinite looping API call ⚠️  

check your network tab in your browsers dev tools

---

When the comments are hidden the comment adder is still visible. Adding a comment when the comments are hidden makes it look like it doesn't do anything

---

When you add a comment it is added to the bottom of the list. Same as above, this is confusing and makes it look like it's not working :( 

---

You can currently submit a comment with no body

---

If the post request fails there is no feedback for the user

---


please put in another `/nchelp pr` when you've fixed the infinite loop!

**Review by dappernerddesigns — APPROVED** [review]
Well done. I like the message to the user if there is a problem. I also like you can't post an empty comment. Could there be a clearer message to the user why it won't post an empty comment? Have a think then merge away. Nice work.

**Pull request merged**